### DISCUSSION TITLE
Add a flow trigger category “ parameter call trigger ”

### DISCUSSION THREAD

**Discussion by hulinux** [directus/directus#16751] [Feature Requests]
At present, it can only be triggered by sending HTTP requests or creating/update data. Sometimes I want to trigger a flow through function calls in Endpoint and get the call result. Can they consider?
like this：const result = (triggerID, params)=>{
.......
return data
}

**Comment by raphourbe** [comment]
Hi,
The solution of using a GET webhook trigger does not work the way you want? https://docs.directus.io/reference/system/flows.html#flow-with-get-webhook-trigger

If you call a Flow triggered by a GET Webhook with some params, you can use them in your Flow. If I use in a Run Script operation with `console.log(data)` after my Event triggered by `http://localhost:8055/flows/trigger/9033763a-cd26-4eee-a309-0af14291622b?param1=hello&param2=world`, it gives me: 

![image](https://user-images.githubusercontent.com/37442071/206398716-630abff1-6dc8-4249-a469-12de7359560e.png)

So you can probably use your Endpoint to call your Flow.

Something like:
```js
const axios = require("axios");

const result = (flow_uuid, params)=>{
  try {
    const flow_response = await axios({
      url: "yourdirectus/flows/trigger/" + flow_uuid + "?param1=" + params[0],
      method: "GET",
    }).then((response) => {
      // whatever you want to do with response
    });
  } catch (err) {
    console.log(err);
  }
}
```

**Comment by hulinux** [comment]
Thank you so much! Yes, at the moment it can only be done in this way, and the effect is as expected, I just want to avoid URLs in program calls, just like calling normal functions to achieve this function.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
